debezium-connector-postgres/src/main/java/io/debezium/connector/postgresql/connection/PostgresReplicationConnection.java

```diff
@@ -521,13 +521,47 @@ public Optional<SlotCreationResult> createReplicationSlot() throws SQLException
         initPublication();
 
         try (Statement stmt = pgConnection().createStatement()) {
+            stmt.setQueryTimeout(toIntExact(connectorConfig.createSlotCommandTimeout()));
             String createCommand = String.format(
                     "CREATE_REPLICATION_SLOT \"%s\" %s LOGICAL %s",
                     slotName,
                     tempPart,
                     plugin.getPostgresPluginName());
             LOGGER.info("Creating replication slot with command {}", createCommand);
-            stmt.execute(createCommand);
+
+            final int maxRetries = connectorConfig.maxRetries();
+            final Duration delay = connectorConfig.retryDelay();
+            int tryCount = 0;
+            while (true) {
+                try {
+                    stmt.execute(createCommand);
+                    break;
+                }
+                catch (SQLException ex) {
+                    // intercept the statement timeout error and retry
+                    if (ex.getMessage().contains("canceling statement due to user request")) {
```

Comment:
On further reflection, i think we should check for `SQLSTATE 55P03`, too. That is `lock_not_available` and is thrown by PostgreSQL if someone has configured server side `lock_timeout` parameter and if this value is lower than the one configured in Debezium, this code can handle this the same way IMHO.

Reply:
@rajdangwal This is definitely good point, @rajdangwal could you please log a follow-up Jira for that as there is one more instance of the same approach in the same class?

@psoo Is the error code unique? I mean is it possible that there are multiple error (error messages) thrown with a the same error code?

Reply:
@jpechane Sorry, saw your comment too late.

The "problem" is indeed the `query_canceled` SQLSTATE. This can also be our backend canceled by a `SIGINT` or `pg_cancel_backend()` by the DBA. I assume pgjdbc calls `cancel()` on the statement handle after receiving the timeout, which does the same. So we can't distinguish whether the DBA cancelled our session or we ran into a timeout.

`lock_not_available` in contrast should be pretty unique: it means the backend couldn't gain the requested lock and timed out.

That said, SQLSTATE is usually pretty well defined what conditions they describe. And i don't think that `query_canceled` is a problem here: we're already trying `MaxRetries` and then giving up. If there is a real problem and the DBA wants to get us out of the way, he should call `pg_terminate_backend()` or SIGTERM us, which gives a `admin_shutdown` (57P01) SQLSTATE.

In summary, i think this is much better than parsing the error message: localized instances would just render our retry logic here useless.

Reply:
Thanks @psoo @jpechane 
Sure, I will log a follow-up Jira for handling the exceptions based on `SQLSTATE` instead of localized error messages.

Reply:
Created: https://issues.redhat.com/browse/DBZ-8127

Will follow-up with a PR.

Reply:
Raised a PR here: https://github.com/debezium/debezium/pull/5763
Please take a look. Thanks.